interoptest/src/javaservice/src/main/java/io/opencensus/interop/ServiceHopper.java

```diff
@@ -0,0 +1,166 @@
+      return TestResponse.newBuilder().setId(id).addStatus(SUCCESS).build();
+    }
+    ServiceHop first = hops.get(0);
+    String host = first.getService().getHost();
+    int port = first.getService().getPort();
+    Spec.Transport transport = first.getService().getSpec().getTransport();
+    Spec.Propagation propagation = first.getService().getSpec().getPropagation();
+    List<ServiceHop> rest = hops.subList(1, hops.size());
+    TestRequest restRequest =
+        TestRequest.newBuilder().setId(id).setName(name).addAllServiceHops(rest).build();
+    try (Scope tagScope = scopeTags(first.getTagsList())) {
+      switch (transport) {
+        case HTTP:
+          switch (propagation) {
+            case B3_FORMAT_PROPAGATION:
+              return httpServiceHop(id, name, host, port, restRequest, B3Format.httpClient);
+            case TRACE_CONTEXT_FORMAT_PROPAGATION:
+              return httpServiceHop(id, name, host, port, restRequest, TcFormat.httpClient);
+            default:
+              return setFailureStatus(id, "Unsupported propagation: " + propagation);
+          }
+        case GRPC:
+          switch (propagation) {
+            case BINARY_FORMAT_PROPAGATION:
+              return grpcServiceHop(id, name, host, port, restRequest);
+            default:
+              return setFailureStatus(id, "Unsupported propagation: " + propagation);
+          }
+        default:
+          return setFailureStatus(id, "Unknown transport: " + transport);
+      }
+    }
+  }
+
+  private static Scope scopeTags(List<Tag> tags) {
+    TagContextBuilder builder = tagger.currentBuilder();
+    for (Tag tag : tags) {
+      builder.put(TagKey.create(tag.getKey()), TagValue.create(tag.getValue()));
+    }
+    return builder.buildScoped();
+  }
+
+  private static TestResponse setFailureStatus(long id, String msg) {
+    CommonResponseStatus status =
+        CommonResponseStatus.newBuilder().setStatus(Status.FAILURE).setError(msg).build();
+    return TestResponse.newBuilder().setId(id).addStatus(status).build();
+  }
+
+  private static TestResponse addSuccessStatus(TestResponse response) {
+    return TestResponse.newBuilder()
+        .setId(response.getId())
+        .addStatus(SUCCESS)
+        .addAllStatus(response.getStatusList())
+        .build();
+  }
+
+  private static TestResponse grpcServiceHop(
+      long id, String name, String host, int port, TestRequest request) {
+    ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port)
+                             // Channels are secure by default (via SSL/TLS).
+                             // For the test, we disable TLS to avoid
+                             // needing certificates.
+                             .usePlaintext(true)
+                             .build();
+    TestExecutionServiceGrpc.TestExecutionServiceBlockingStub blockingStub =
+        TestExecutionServiceGrpc.newBlockingStub(channel);
+    TestResponse response = blockingStub.test(request);
```

Comment:
would it block forever or would it timeout? If timeout occurs then the response should be a failure.

Reply:
That's an interesting point - it's probably better to set a timeout and return a failure.  I've added a 1 second timeout here and below but we should be consistent across languages about a reasonable timeout period for these tests.

Consider adding that to the interoperability_test.proto.